README.md

```diff
@@ -18,7 +26,137 @@ The Nickel library contains in-code documentation that can be leveraged by the
 - `nickel query -f nix.ncl lib.nix_string_hack` will show the documentation of a
     specific symbol, here `lib.nix_string_hack`.
 
-### Examples
+## Usage
+
+The [`example/nix-shell`](examples/nix-shell/) illustrates how to use
+`nickel-nix` to writer a simple `hello` shell.
+
+To try it, enter the `example/nix-shell` directory, and run:
+
+```
+$ nix develop --impure
+Development shell
+Hello, world!
+```
+
+This is an hello from the Nickel world!
+
+Please refer to the [example's README](examples/nix-shell/README.md) for more
+details.
+
+More examples of varied Nix derivations are to come.
+
+## Why using Nickel for Nix ?
+
+There are already resources on what is Nickel, and why it can make for better
+user experience for Nix[^1]. In particular, Nickel adds validations capabilities
+thanks to its type system and contract system. As a simple illustration, let's
+make a stupid error in our Nix version of the `hello` shell,
+[`examples/nix-shell/shell.nix`](examples/nix-shell/shell.nix), by replacing the
+list of `packages` by a string:
+
+```nix
+-    packages = [ pkgs.hello ];
++    packages = "pkgs.hello";
+```
```

Comment:
I think it would be better to provide attributes to the flake which do use such faulty implementations. Instead of asking people to make changes. What do you think?

Reply:
Yup, good idea.